latch/ldata/path.py

```diff
@@ -0,0 +1,171 @@
+from latch_cli.utils import get_auth_header, urljoins
+from latch_cli.utils.path import is_remote_path
+
+
+class LPath:
+    def __init__(self, path: str):
+        if not is_remote_path(path):
+            raise ValueError(f"Invalid LPath: {path} is not a Latch path")
+        self.path = path
+        self._node_id = None
+
+    @property
+    def node_id(self) -> str:
+        if self._node_id is None:
+            self._node_id = get_node_data(self.path).data[self.path].id
+        return self._node_id
+
+    @property
+    def exists(self) -> bool:
+        try:
+            node_data = get_node_data(self.path).data[self.path]
+        except FileNotFoundError:
+            return False
+        return not node_data.removed
+
+    @property
+    def name(self) -> str:
+        return get_node_data(self.path).data[self.path].name
+
+    @property
+    def type(self) -> LDataNodeType:
+        return get_node_data(self.path).data[self.path].type
+
+    def is_dir(self) -> bool:
+        return self.type in {
+            LDataNodeType.dir,
+            LDataNodeType.account_root,
+            LDataNodeType.mount,
+        }
+
+    @property
+    def size(self) -> float:
+        metadata = get_node_metadata(self.node_id)
+        return metadata.size
+
+    @property
+    def content_type(self) -> str:
+        metadata = get_node_metadata(self.node_id)
+        return metadata.content_type
+
+    def iterdir(self) -> Generator[Path, None, None]:
+        if not self.is_dir():
+            raise ValueError(f"Not a directory: {self.path}")
+        data = execute(
+            gql.gql("""
+            query LDataChildren($argPath: String!) {
+                ldataResolvePathData(argPath: $argPath) {
+                    finalLinkTarget {
+                        childLdataTreeEdges(filter: { child: { removed: { equalTo: false } } }) {
+                            nodes {
+                                child {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"argPath": self.path},
+        )["ldataResolvePathData"]
+
+        if data is None:
+            raise ValueError(f"No directory found at path: {self.path}")
+
+        for node in data["finalLinkTarget"]["childLdataTreeEdges"]["nodes"]:
+            yield urljoins(self.path, node["child"]["name"])
+
+    def rmr(self) -> None:
+        execute(
+            gql.gql("""
+            mutation LDataRmr($nodeId: BigInt!) {
+                ldataRmr(input: { argNodeId: $nodeId }) {
+                    clientMutationId
+                }
+            }
+            """),
+            {"nodeId": self.node_id},
+        )
+
+    def copy(self, dst: Union["LPath", str]) -> None:
+        remote_copy(self.path, str(dst))
+
+    def upload(self, src: Path, progress=Progress.tasks, verbose=False) -> None:
```

Comment:
```suggestion
    def upload(self, src: Path, *, progress=Progress.tasks, verbose=False) -> None:
```

imo drop the progress stuff from the public API - just leave a `show_progress_bar` boolean

it should also default to `False` in workflows probably since they don't have a TTY